src/Core/Framework/Routing/Event/MaintenanceAllowedClientEvent.php

```diff
@@ -0,0 +1,44 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Framework\Routing\Event;
+
+use Shopware\Core\Framework\Log\Package;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Contracts\EventDispatcher\Event;
+
+#[Package('core')]
+class MaintenanceAllowedClientEvent extends Event
```

Comment:
I think we can rename this to something like `MaintenanceModeRequestEvent` what do you think?